libr/include/r_anal.h

```diff
@@ -215,6 +215,49 @@ enum {
 	R_ANAL_DIFF_TYPE_UNMATCH = 'u'
 };
 
+typedef struct enum_case {
+	char *name;
+	const int val;
+} RAnalEnumCase;
+
+typedef struct struct_member {
+	char *name;
+	char *type;
+	const int offset;
+} RAnalStructMember;
+
+typedef struct union_member {
+	char *name;
+	char *type;
+} RAnalUnionMember;
+
+typedef enum {
```

Comment:
what about naming it RAnalType. instead of the BaseType? do that conflicts? why is KIND needed? this enum looks too long

Reply:
About the name, see https://hackmd.io/kKvaIscWQk6_zFLYcLYWnw
There must be a clear distinction between "C-types" and "base types" with the current design of the database. The "Kind" comes from the typedefd name of the enum itself.

Reply:
> About the name, see https://hackmd.io/kKvaIscWQk6_zFLYcLYWnw

Why should the Ghidra solution be "easier to do it if the above solution with base types and c types is implemented already"?

Reply:
Because then the entire code will already use C structs everywhere as opposed to strings. Refactoring these C structs into a common pool would be easier than doing the same from strings.